### PR TITLE
Refactor/creation preview itinerary

### LINKED ISSUE #332: ISSUE#317 Refator preview creation screen

### PR DESCRIPTION
**Pull request by gunnm0**
closes #332 

# Changed creation preview banner screen
Screenshot from 2024-06-02 15-05-59

## Banner view changes
added dummy banners to simulate the home screen in preview of itinerary creation

## Additional changes

- fixed bug where you could delete itinerary in creation mode (made the app crash)
- added comments
- removed useless parameters in functions
- changed some composables to use the color scheme
- (maybe other things I forgot)

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt:1]: Good code, but i still don't really understand why do we need to add those dummy itineraries.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt [hunk 1]
```diff
@@ -307,3 +318,47 @@ fun BannerLikeButton(
...
+  val tag1 = space.repeat(Random.nextInt(5, 11))
+  val tag2 = space.repeat(Random.nextInt(5, 11))
+  Card(
+      colors =
+          CardDefaults.cardColors(
+              containerColor = decreaseAlpha(MaterialTheme.colorScheme.tertiary, 2f),
+          ),
+      shape = RoundedCornerShape(9.dp),
+      modifier = Modifier.testTag("${TestTags.DUMMY_BANNER}_${uid}"),
+      onClick = {}) {
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:338]: Is there a purpose for the onClick attribute ? if not then can you remove it ?

File: app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt:1]: Nice spot

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:1]: Good comments

File: app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt [hunk 2]
```diff
@@ -392,7 +397,7 @@ private fun PreviewItineraryBannerMaximized(
           Spacer(modifier = Modifier.height(20.dp))
           val coroutineScope = rememberCoroutineScope()
           val uid = profileViewModel.getUserUid()
-          if (uid == itinerary.userUid) {
+          if (uid == itinerary.userUid && !creationMode) {
```
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt:400]: good catch

File: app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt
Comment (engjellismaili) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt:1]: Lgtm

**Review by engjellismaili — APPROVED** [review]
Overall lgtm, clean code and good comments, even if at the end I don't full understand the utility of those dummy itineraries.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt (hunk 2 above)
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/map/PreviewItineraryScreen.kt:400]: Nice addition with the `creationMode` boolean to make sure that the delete itinerary button does not appear while creating the itinerary.

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt [hunk 3]
```diff
@@ -307,3 +318,47 @@ fun BannerLikeButton(
...
+ */
+@Composable
+fun DummyBanner(uid: String) {
+  val space: String = " "
+  val tag1 = space.repeat(Random.nextInt(5, 11))
+  val tag2 = space.repeat(Random.nextInt(5, 11))
+  Card(
+      colors =
+          CardDefaults.cardColors(
+              containerColor = decreaseAlpha(MaterialTheme.colorScheme.tertiary, 2f),
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:334]: Is it possible to decrease the Alpha of the dummy itineraries even more? I still feel like they take too much attention away from the actual itinerary banner we created

File: app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/TestTags.kt:1]: maybe make sure that the value that you give is consistent with the previous test tags

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/steps/CreationStepPreviewBanner.kt:1]: the dummy itineraries give a more realistic feel of how my itinierary will look like amongst the others, great job

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt [hunk 4]
```diff
@@ -307,3 +318,47 @@ fun BannerLikeButton(
...
+  Surface(
+      modifier = Modifier.fillMaxSize(),
+      color = decreaseAlpha(MaterialTheme.colorScheme.surfaceVariant, 2f)) {}
+}
+
+/** @brief divided the color alpha by alphaDecrease, returns a new Color */
+fun decreaseAlpha(color: Color, alphaDecrease: Float): Color {
+  val originalAlpha = color.alpha
+  val newAlpha = (originalAlpha / alphaDecrease).coerceIn(0f, 1f)
+  return Color(color.red, color.green, color.blue, newAlpha)
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:363]: lgtm

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:1]: great changes to the colors for your different elements

File: app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt [hunk 5]
```diff
@@ -249,16 +260,16 @@ fun Price(price: Float) {
 
 /** @brief write the style of the likes attribute */
 @Composable
-fun Likes(numLikes: Int) {
+fun Likes(numLikes: String) {
```
Comment (ThomasPFMuller) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/itinerary/ItineraryBanner.kt:263]: Why did you change these to strings? Does the filtering by time and price still work?

File: app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt
Comment (imaneb942) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/navigation/ItineraryCreationNavigationMenu.kt:1]: nice clean up, this is what we want

**Review by ThomasPFMuller — COMMENTED** [review]
Overall nice changes, please answer my doubts for an approval

**Review by imaneb942 — APPROVED** [review]
the changes look great and the additions whether it be in you comments or refactoring are interesting, good job
